src/rust/ensogl/lib/core/src/application/frp.rs

```diff
@@ -207,7 +207,7 @@ macro_rules! define_endpoints {
             NORMALIZED
             $([$($global_opts)*])?
 
-            Input  { $([$($input_opts)*])?
+            Input  { $($([$($input_opts)*])?)?
```

Comment:
Because it doesnt work otherwise. Look at the definition. Definition uses double optional expansion, so the body has to use one as well. It was invalid in the past!